fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -401,6 +402,7 @@ private OptExpression logicalRuleRewrite(ConnectContext connectContext,
         // After this rule, we shouldn't generate logical project operator
         ruleRewriteIterative(tree, rootTaskContext, new MergeProjectWithChildRule());
 
+        ruleRewriteOnlyOnce(tree, rootTaskContext, new OuterJoinAddRedundantTopNRule());
```

Comment:
execute it before `MergeProjectWithChildRule`

Reply:
If before `MergeProjctWithChildRule`, it may need two patterns. I had discussed with stephen,  the check process has ensure all order by cols are from the child of join. The added topN operator also without setting projection. So after the rule it OK? Has any other problem here?

Reply:
OK, just few supported scenarios now